### DISCUSSION TITLE
There is "otThreadSetEnabled" but no "otThreadGetState"?

### DISCUSSION THREAD

**Discussion by semi10** [openthread/openthread#8955] [Q&A] [answered]
There is `otThreadSetEnabled` function but it seems that there is no way to check if OpenThread is enabled "`otThreadGetState`".
It is possible to store some flag for the sate but it seems odd to me that there is no built in function (or is there?)

As you know  it is required to disable OT before changing the channel mask. My intention / use case is to change channel mask without changing the OpenThread final state (if it was enabled keep it enabled and vise versa)...

Is there already built in function to verify current OpenThread state? 

Thank you in advance!

**Comment by jwhui** [comment, marked as answer]
See [otThreadGetDeviceRole](https://openthread.io/reference/group/api-thread-general#otthreadgetdevicerole)

**Reply by semi10** [reply]
Thx! Exactly what I was looking for.
